pandas/core/frame.py

```diff
@@ -1266,7 +1266,7 @@ def from_dict(cls, data, orient="columns", dtype=None, columns=None) -> "DataFra
             if len(data) > 0:
                 # TODO speed up Series case
                 if isinstance(list(data.values())[0], (Series, dict)):
-                    data = _from_nested_dict(data)
+                    data = lib.from_nested_dict(data)
```

Comment:
If you change the above to

```python
data = dict(data) if not type(data) is dict else data  # convert OrderedDict
data = lib.from_nested_dict(data)
```

you can change the function interface in ``lib`` to ``def from_nested_dict(dict data) -> dict:``, which will simplify things and make it faster too.